src/test/java/at/ac/tuwien/kr/alpha/AlphaTest.java

```diff
@@ -0,0 +1,200 @@
+package at.ac.tuwien.kr.alpha;
+
+import at.ac.tuwien.kr.alpha.common.AnswerSet;
+import at.ac.tuwien.kr.alpha.common.AnswerSetBuilder;
+import at.ac.tuwien.kr.alpha.common.Program;
+import at.ac.tuwien.kr.alpha.common.Rule;
+import at.ac.tuwien.kr.alpha.common.atoms.BasicAtom;
+import at.ac.tuwien.kr.alpha.common.atoms.ExternalAtom;
+import at.ac.tuwien.kr.alpha.common.predicates.Predicate;
+import at.ac.tuwien.kr.alpha.common.predicates.ExternalMethodPredicate;
+import at.ac.tuwien.kr.alpha.common.terms.ConstantTerm;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+
+public class AlphaTest {
+	private static int invocations;
+
+	@at.ac.tuwien.kr.alpha.Predicate
+	public static boolean isOne(int term) {
+		invocations++;
+		return term == 1;
+	}
+
+	@at.ac.tuwien.kr.alpha.Predicate
+	public static boolean isFoo(Integer a) {
+		return a == 0xF00;
+	}
+
+	@at.ac.tuwien.kr.alpha.Predicate
+	public static boolean thinger(Thingy thingy) {
```

Comment:
I believe I just now understand what this enables us to do. Maybe that's worth a paper!

Reply:
Do you have any publication outlets (conferences etc.) in mind that would be interested in these kind of things? I'd like to get a feel for what is published there.

Reply:
There might be some, it depends on the actual use cases we have. We could discuss that directly by email.